_inc/lib/class.core-rest-api-endpoints.php

```diff
@@ -12,6 +12,12 @@
 	exit;
 }
 
+// This would fatal on versions below WP 4.4
+global $wp_version;
+if ( ! function_exists( 'rest_api_init' ) || version_compare( $wp_version, '4.4-z', '<=' ) ) {
+	wp_die( __( 'Minimum WordPress version 4.4 required to unlock the power of the WP REST API', 'jetpack' ) );
+}
```

Comment:
That's an interesting idea... @beaulebens brings up a point in a comment below about if we should be supporting old versions at all.  I think I'd be fine with that. It would also make this logic a lot more simple, as we'd only have to account for noJS.